src/Microsoft.Data.Analysis/DataFrameColumnCollection.cs

```diff
@@ -175,5 +175,278 @@ protected override void ClearItems()
             }
         }
 
+        /// <summary>
+        /// Searches for an <see cref="PrimitiveDataFrameColumn{T}"/> with the specified <paramref name="name"/> and attempts to return it as an <see cref="PrimitiveDataFrameColumn{T}"/>. If <typeparamref name="T"/> does not match <see cref="PrimitiveDataFrameColumn.DataType"/>, an exception is thrown.
+        /// </summary>
+        /// <param name="name">The name of the column</param>
+        /// <returns><see cref="PrimitiveDataFrameColumn{T}"/>.</returns>
+        /// <exception cref="ArgumentException">Throws this exception if types don't match</exception>
+        public PrimitiveDataFrameColumn<T> GetPrimitiveColumn<T>(string name)
+            where T : unmanaged
+        {
+            DataFrameColumn column = this[name];
+            if (column is PrimitiveDataFrameColumn<T> ret)
+            {
+                return ret;
+            }
+
+            throw new ArgumentException(string.Format(Strings.BadColumnCast, column.DataType, typeof(T)), nameof(T));
```

Comment:
I don't know if it's worth it...

Perf-wise this will prevent this method from being inlined. Not only the `throw`, but also the construction of the exception-message.
So if relevant, move the "throwing" to a ThrowHelper or a local function. Then this (tiny) method will be inlined into the caller.

Reply:
Filed https://github.com/dotnet/corefxlab/issues/2881 to address this.